File: src/frontend/lib/dto/Meeting/join_meeting_response_dto.dart
class JoinMeetingResponseDto {
  final String appId;
  final String channelName;
  final String token;
  final int uid;
  final String appointmentId;
  final String role;

  JoinMeetingResponseDto({
    required this.appId,
    required this.channelName,
    required this.token,
    required this.uid,
    required this.appointmentId,
    required this.role,
  });

  factory JoinMeetingResponseDto.fromJson(Map<String, dynamic> json) {
    return JoinMeetingResponseDto(
      appId: json['appId'] as String? ?? '',
      channelName: json['channelName'] as String? ?? '',
      token: json['token'] as String? ?? '',
      uid: (json['uid'] as num?)?.toInt() ?? 0,
      appointmentId: json['appointmentId'] as String? ?? '',
      role: json['role'] as String? ?? 'publisher',
    );
  }
}

File: src/backend/src/main/java/com/bouh/backend/model/Dto/Meeting/JoinMeetingResponseDto.java
package com.bouh.backend.model.Dto.Meeting;

public class JoinMeetingResponseDto {
    private String appId;
    private String channelName;
    private String token;
    private Integer uid;
    private String appointmentId;
    private String role;

    public JoinMeetingResponseDto() {}

    public JoinMeetingResponseDto(
            String appId,
            String channelName,
            String token,
            Integer uid,
            String appointmentId,
            String role) {
        this.appId = appId;
        this.channelName = channelName;
        this.token = token;
        this.uid = uid;
        this.appointmentId = appointmentId;
        this.role = role;
    }

    public String getAppId() { return appId; }
    public void setAppId(String appId) { this.appId = appId; }

    public String getChannelName() { return channelName; }
    public void setChannelName(String channelName) { this.channelName = channelName; }

    public String getToken() { return token; }
    public void setToken(String token) { this.token = token; }

    public Integer getUid() { return uid; }
    public void setUid(Integer uid) { this.uid = uid; }

    public String getAppointmentId() { return appointmentId; }
    public void setAppointmentId(String appointmentId) { this.appointmentId = appointmentId; }

    public String getRole() { return role; }
    public void setRole(String role) { this.role = role; }
}
